File: android/app/src/main/kotlin/com/example/flutter_body/actions/RehabActionCallback.kt
package com.example.flutter_body.actions

interface RehabActionCallback {
    fun updateUI(title: String? = null, instruction: String? = null, feedback: String? = null, repCount: String? = null, accuracy: String? = null)
    fun speak(text: String, isUrgent: Boolean = false)
    fun speakCount(count: Int)
    fun onTrainingComplete(report: TrainingReport)
    fun setGuideLineVisible(visible: Boolean)
    fun setPinchGuideEnabled(visible: Boolean)
    fun updateProgress(progress: Float, speedState: Int)
    fun setSkeletonMode(mode: String = "FULL_BODY")
}

File: lib/actions/rehab_action_callback.dart
// lib/actions/rehab_action_callback.dart

abstract class RehabActionCallback {
  void onFeedbackChanged(String feedback, String instruction);

  void onStatsChanged({
    int? repCount,
    double? accuracy,
    double? progress,
    int? speedState,
  });

  /// 翻掌專用：倒數狀態變化
  void onCountdownChanged({
    required bool isCountingDown,
    required int seconds,
    required bool isDone,
  });

  void onTrainingComplete({
    required int repCount,
    required int durationSeconds,
    required List<String> mistakeLogs,
  });
}